app/dojo.py

```diff
@@ -0,0 +1,205 @@
+from random import choice
+from app.rooms import Room, Office, LivingSpace
+from app.person import Person, Fellow, Staff
+
+
+class Dojo():
+
+	def __init__(self):
+		self.all_rooms = []
+		self.offices = []
+		self.living_spaces = []
+		self.all_people = []
+		self.fellows = []
+		self.staffs = []
+		self.waiting_to_allocate_living = []
+		self.waiting_to_allocate_office = []
+
+	def create_room(self, room_type, room_names):
+		""" function to create a unique room space """
+		msg = ''
+		for name in room_names:
+			room_name = name.lower()
+			room_type = room_type.lower()
+
+			found = False
+			for room in self.all_rooms:
+				if room_name == room.room_name:
+					found = True
+			if found:
+				msg += "Room with name " + room_name + " already exists \n"
+			else:
+				if room_type == 'office':
+					room = Office(room_name)
+					self.offices.append(room)
+					self.all_rooms.append(room)
+					msg += "Office " + room_name + " added successfully \n"
+				elif room_type == 'living':
+					room = LivingSpace(room_name)
+					self.living_spaces.append(room)
+					self.all_rooms.append(room)
+					msg += "Living space " + room_name + " added successfully\n"
+				else:
+					msg += "Wrong room type \n"
+		return msg
+
+
+	def add_person(self, first_name, last_name, category, accomodation="N"):
+		""" function to add a person """
+		if isinstance(first_name, str) and isinstance(last_name, str):
+			if category.lower() == "fellow":
+				person = Fellow(first_name, last_name, accomodation)
+				person.id = int(len(self.all_people)+1)
+				self.fellows.append(person)
+				self.all_people.append(person)
+				self.waiting_to_allocate_office.append(person)
+				if accomodation.upper() == 'Y':
+					self.waiting_to_allocate_living.append(person)
+					return "Fellow " + person.first_name + " " + \
+					 person.last_name + " added successfully with id " \
+					 + str(person.id) + ", " + str(self.allocate_office(person)) \
+					 + " and " + str(self.allocate_living(person))
+				return "Fellow " + person.first_name + " " + person.last_name\
+				 + " added successfully with id " + str(person.id) + " and " \
+				+ str(self.allocate_office(person))
```

Comment:
Since the fellow can be allocated to both living and office, I think this script would not allocate an office if the accommodation option is Y. I think appending the messages then returning them at the end would be better.

I think your output is going to be odd. if no office spaces are available and you try add a fellow, the output will read:
Fellow x y added successfully with id 0 and Sorry ! No offices with space available.